### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by Danwhy**
Adds profile page and links it with redcap, enabling users to see and update their info
ref #35

### CODE REVIEW

File: web/controllers/account_controller.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    |> @redcap_api.save_record
+    |> case do
+      {:ok, %HTTPoison.Response{status_code: 200}} ->
+        conn
+        |> put_flash(:info, "Profile Updated")
+        |> redirect(to: account_path(conn, :index))
+      _ ->
+        conn
+        |> put_flash(:error, "Something went wrong")
+        |> redirect(to: account_path(conn, :index))
```
Comment (samhstn) [inline web/controllers/account_controller.ex:32]: Love this 👍

File: web/controllers/account_controller.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  @redcap_api Application.get_env(:research_resource, :redcap_api)
+
+  plug :authenticate_user when action in [:index]
+
+  def index(conn, _params) do
+    user_details = Enum.to_list @redcap_api.get_user_data(conn.assigns.current_user.ttrrid)
+    # Turn user data from a map to a keyword list
+    |> Enum.map(fn(detail) ->
+      {key, val} = detail
+      {atom_key, val} = {String.to_atom(key), val}
```
Comment (samhstn) [inline web/controllers/account_controller.ex]: Maybe this bit can be tidied up a little
```elixir
Enum.map(fn {key, val} -> {String.to_atom(key), val} end)
```

**Review by samhstn — CHANGES_REQUESTED** [review]

**Review by samhstn — CHANGES_REQUESTED** [review]
Minor syntax point, but all good

So nice to read through @Danwhy 👍

**Comment by Danwhy** [comment]
Thanks @shouston3, I've added your suggestion

File: web/controllers/account_controller.ex [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  alias ResearchResource.Redcap.RedcapHelpers
+
+  @redcap_api Application.get_env(:research_resource, :redcap_api)
+
+  plug :authenticate_user when action in [:index]
+
+  def index(conn, _params) do
+    user_details = Enum.to_list @redcap_api.get_user_data(conn.assigns.current_user.ttrrid)
+    # Turn user data from a map to a keyword list
+    |> Enum.map(fn {key, val} -> {String.to_atom(key), val} end)
```
Comment (samhstn) [inline web/controllers/account_controller.ex]: I'm fine if you don't want to implement this, but my personal preference would be:
```elixir
user_details =
  conn.assigns.current_user.ttrrid
  |> @redcap_api.get_user_data
  |> Enum.to_list
  |> Enum.map(fn {key, val} -> {String.to_atom(key), val} end)
```

Just so it is consistent how we are applying the functions

But if you would prefer to leave this, that's good with me

**Review by samhstn — APPROVED** [review]
Looks good, over to you @SimonLab

File: web/controllers/account_controller.ex (hunk 3 above)
Comment (Danwhy) [reply web/controllers/account_controller.ex]: No I agree, your way looks much nicer

**Review by SimonLab — APPROVED** [review]
Looks good, thanks @shouston3 for the review

**Pull request merged**